core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -327,6 +328,15 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
     }
     for (int i = rank_start; i != rank_end; i += increment) {
       const index_type length = m_upper[i] - m_lower[i];
+
+      if (m_upper[i] < m_lower[i]) {
+        std::stringstream msg;
+        msg << "Kokkos::MDRangePolicy bounds error: The lower bound ("
+            << m_lower[i] << ") is greater than its upper bound (" << m_upper[i]
+            << ") in rank " << i + 1 << ".";
+        Kokkos::abort(msg.str().c_str());
```

Comment:
can we avoid pulling the sstream header into this and just use a regular string?

Reply:
Converted to a string in [341b464](https://github.com/kokkos/kokkos/pull/6617/commits/341b464d38638c07fb4767c4becb35ab78859dbd).